moveit_core/kinematic_constraints/src/kinematic_constraint.cpp

```diff
@@ -607,23 +650,26 @@ ConstraintEvaluationResult OrientationConstraint::decide(const moveit::core::Rob
   if (mobile_frame_)
   {
     // getFrameTransform() returns a valid isometry by contract
-    Eigen::Matrix3d tmp = state.getFrameTransform(desired_rotation_frame_id_).linear() * desired_rotation_matrix_;
+    const Eigen::Vector3d desired_rpy =
+        CalcEulerAngles(state.getFrameTransform(desired_rotation_frame_id_).linear() * desired_rotation_matrix_);
     // getGlobalLinkTransform() returns a valid isometry by contract
-    Eigen::Isometry3d diff(tmp.transpose() * state.getGlobalLinkTransform(link_model_).linear());  // valid isometry
-    xyz = diff.linear().eulerAngles(0, 1, 2);
-    // 0,1,2 corresponds to XYZ, the convention used in sampling constraints
+    const Eigen::Vector3d current_rpy =
+        CalcEulerAngles(state.getGlobalLinkTransform(link_model_).linear());  // valid isometry
+    xyz = current_rpy - desired_rpy;
   }
   else
   {
-    // diff is valid isometry by construction
-    Eigen::Isometry3d diff(desired_rotation_matrix_inv_ * state.getGlobalLinkTransform(link_model_).linear());
-    xyz = diff.linear().eulerAngles(0, 1, 2);  // 0,1,2 corresponds to XYZ, the convention used in sampling constraints
+    const Eigen::Vector3d desired_rpy = CalcEulerAngles(desired_rotation_matrix_);
+    const Eigen::Vector3d current_rpy = CalcEulerAngles(state.getGlobalLinkTransform(link_model_).linear());
+    xyz = current_rpy - desired_rpy;
   }
-
   // Account for angle wrapping
-  xyz(0) = std::min(fabs(xyz(0)), boost::math::constants::two_pi<double>() - fabs(xyz(0)));
-  xyz(1) = std::min(fabs(xyz(1)), boost::math::constants::two_pi<double>() - fabs(xyz(1)));
-  xyz(2) = std::min(fabs(xyz(2)), boost::math::constants::two_pi<double>() - fabs(xyz(2)));
+  xyz = xyz.unaryExpr([](const double angle) {
+    const double normalized_angle = std::fmod(std::abs(angle), boost::math::constants::two_pi<double>());
```

Comment:
A small nit that was already in the existing code, but would you mind using `namespace bmc = boost::math::constants` or `M_PI` depending on your stance on  #2285. I think the extremely long name for `pi` really makes it harder to reason about the code.

(I know this is a code base wide issue, so feel free to ignore it in this PR.)